python/python/lancedb/query.py

```diff
@@ -1691,6 +1697,17 @@ def nearest_to(
                 self._inner.nearest_to(AsyncQuery._query_vec_to_array(query_vector))
             )
 
+    def rerank(
+        self,
+        reranker: Reranker = RRFReranker(),
+    ) -> AsyncFTSQuery:
+        if reranker and not isinstance(reranker, Reranker):
+            raise ValueError("reranker must be an instance of Reranker class.")
+
+        self._reranker = reranker
+
+        return AsyncFTSQuery(self._inner.rerank(reranker))
```

Comment:
I'm not sure this method is needed. If the user wanted to do FTS search with a reranker, they could probably do something like:
```py
table.query()
  .nearest_to_text("foo") # creates an AsyncFTSQuery
  .reranker(RRFReranker())
```